android/src/main/java/com/blueshift/reactnative/BlueshiftReactNativeModule.java

```diff
@@ -353,6 +363,106 @@ public static void processBlueshiftPushUrl(Intent intent) {
         }
     }
 
+    // MOBILE INBOX
+    @ReactMethod
+    void getInboxMessages(Callback callback) {
+        if (callback != null) {
+            BlueshiftInboxManager.getMessages(getReactApplicationContext(), blueshiftInboxMessages -> {
+                if (blueshiftInboxMessages == null || blueshiftInboxMessages.isEmpty()) {
+                    BlueshiftLogger.d(TAG, "No messages found inside Mobile Inbox.");
+
+                    // return empty list
+                    HashMap<String, Object> messages = new HashMap<>();
+                    messages.put("messages", new ArrayList<>());
+                    callback.invoke(hashMapToWritableMap(messages));
+                } else {
+                    ArrayList<HashMap<String, Object>> messageList = new ArrayList<>();
+
+                    for (BlueshiftInboxMessage message : blueshiftInboxMessages) {
+                        messageList.add(message.toHashMap());
+                    }
+
+                    HashMap<String, Object> messages = new HashMap<>();
+                    messages.put("messages", messageList);
+                    callback.invoke(hashMapToWritableMap(messages));
+                }
+            });
+        }
+
+    }
+
+    @ReactMethod
+    void showInboxMessage(ReadableMap readableMap) {
+        HashMap<String, Object> map = toHashMap(readableMap);
+        if (map != null) {
+            BlueshiftInboxMessage message = BlueshiftInboxMessage.fromHashMap(map);
+            BlueshiftInboxManager.displayInboxMessage(message);
+        } else {
+            Log.d(TAG, "showInboxMessage: No message found to display.");
+        }
+    }
+
+    @ReactMethod
+    void deleteInboxMessage(ReadableMap readableMap, Callback callback) {
+        HashMap<String, Object> map = toHashMap(readableMap);
+        if (map != null) {
+            BlueshiftInboxMessage message = BlueshiftInboxMessage.fromHashMap(map);
+            BlueshiftInboxManager.deleteMessage(getReactApplicationContext(), message, status -> {
+                if (status) {
+                    callback.invoke(true);
+                } else {
+                    callback.invoke(false);
```

Comment:
Incase of failed delete we should send the reason why it failed, the way we send it in the native sdks.